src/push_update.rs

```diff
@@ -0,0 +1,55 @@
+use crate::util::Binding;
+use crate::{raw, Oid};
+use std::marker;
+use std::str;
+
+/// Represents an update which will be performed on the remote during push.
+pub struct PushUpdate<'a> {
+    raw: *const raw::git_push_update,
+    _marker: marker::PhantomData<&'a raw::git_push_update>,
```

Comment:
I don't think you need the `PhantomData` here: it signals that dropping a `PushUpdate` may drop a `raw::git_push_update`, but:

- you didn't implement `Drop` for `PushUpdate` (and you don't need to)
- libgit2 will take care of freeing the `raw::git_push_update` structure

Unless there is another reason you need the `PhantomData` here, you might want to remove this field.